include/uinputdevice.h

```diff
@@ -39,11 +39,13 @@ class UinputDevice
 };
 
 void UinputDevice::open(std::string const& path) {
+  destroy();
   _fd = ::open(path.data(), O_WRONLY | O_NONBLOCK);
 }
 
 UinputDevice::UinputDevice(std::string const& path, unsigned int bus, std::string const& name, unsigned int vendor, unsigned int product, unsigned int version, std::vector<PossibleEvent> const& possibleEvents, std::vector<AbsoluteAxisCalibrationData> const& absoluteAxesCalibrationData)
 {
+  _fd = 0;
```

Comment:
...but this renders it useless. `destroy()` does nothing if you zero `_fd` beforehand.